mlir/lib/Dialect/SCF/Transforms/LoopSpecialization.cpp

```diff
@@ -166,6 +167,52 @@ static LogicalResult peelForLoop(RewriterBase &b, ForOp forOp,
   return success();
 }
 
+static void rewriteVectorReadWriteToLoadStore(RewriterBase &b, Operation *op) {
+  b.setInsertionPoint(op);
+  if (auto write = dyn_cast<vector::TransferWriteOp>(op)) {
+    b.replaceOpWithNewOp<vector::StoreOp>(
```

Comment:
How about simply setting the `in_bounds` attribute to "true"? I think we already have canonicalization/lowering patterns that would then lower the op to `vector.load/store`.

Reply:
I didn't notice there is In_bounds attribute. Thanks for the tips.